types/next/test/next-dynamic-tests.tsx

```diff
@@ -17,26 +17,42 @@ const LoadingComponent: React.StatelessComponent<LoadingComponentProps> = ({
 }) => <p>loading...</p>;
 
 // 1. Basic Usage (Also does SSR)
-const DynamicComponent = dynamic(asyncComponent);
+const DynamicComponent = dynamic(Promise.resolve(MyComponent));
 const dynamicComponentJSX = <DynamicComponent foo="bar" />;
 
+// 1.1 Basic Usage (Loader function, module shape with 'export = Component' / 'module.exports = Component')
+const DynamicComponent2 = dynamic(() => Promise.resolve(MyComponent));
+const dynamicComponent2JSX = <DynamicComponent2 foo="bar" />;
+
+// 1.2 Basic Usage (Loader function, module shape with 'export default Component')
+const DynamicComponent3 = dynamic(() => Promise.resolve({ default: MyComponent }));
+const dynamicComponent3JSX = <DynamicComponent3 foo="bar" />;
+
+// TODO: Work with module shape 'export { Component }'
+
 // 2. With Custom Loading Component
-const DynamicComponentWithCustomLoading = dynamic(asyncComponent, {
+const DynamicComponentWithCustomLoading = dynamic(import('./imports/with-default'), {
+    loading: LoadingComponent
+});
+const dynamicComponentWithCustomLoadingJSX = <DynamicComponentWithCustomLoading foo />;
+
+// 2.1. With Custom Loading Component (() => import('') syntax)
```

Comment:
if we do https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33163/files/02ba783ed3c98ecc0c625474d3d186fa810e367c#r258041360 this is already tested? Group 2. is testing the options. If we want to be really complete than we should have `2.1`, `2.2`, `3.1`, `3.2`, `4.1` and `4.2` for all the possibilities.